RUFAS/report_generator.py

```diff
@@ -271,3 +296,339 @@ def _prepare_report_data(
+                "message": f"Error generating report: {report_name}: {e}",
+                "info_map": info_map,
+            }
+            event_logs.append(error_event)
+
+        return event_logs
+
+    def _generate_unique_report_name(self, filter_content: Dict[str, Any]) -> str:
+        """
+        Generates a unique name for the report based on the filter content.
+
+        Parameters
+        ----------
+        filter_content : Dict[str, Any]
+            The filter content for the report.
+
+        Returns
+        -------
+        str
+            If there is no `name` key in the filter content, the name is set to `untitled_<timestamp>`.
+            If the name is already present in the list of previously generated reports, the timestamp is appended
+            to the name to make it unique.
+            Otherwise, the name is set to the value of the `name` key in the filter content.
+        """
+
+        base_name = filter_content.get("name", f"untitled_{Utility.get_timestamp(True)}")
+
+        if base_name in self.reports:
+            base_name = f"{base_name} {Utility.get_timestamp(True)}"
+
+        return base_name
+
+    def _check_for_missing_references(self, references: List[str]) -> None:
+        """
+        Checks if all the referenced reports are present.
+
+        Parameters
+        ----------
+        references : List[str]
+            The list of report references to check.
+
+        Raises
+        ------
+        KeyError
+            If any of the report references are missing.
+        """
+
+        missing_references = [ref for ref in references if ref not in self.reports]
+        if missing_references:
+            raise KeyError(f"Missing referenced reports: {', '.join(missing_references)}")
+
+    @staticmethod
+    def _generate_single_report(filtered_pool: Dict[str, Dict[str, List[Any]]],
+                                filter_content: Dict[str, Any]) -> List[float]:
+        """
+        Generates a report based on filtered data and aggregation criteria, including scalar operations and constants.
+
+        Parameters
+        ----------
+        filtered_pool : Dict[str, Dict[str, List[Any]]]
+            The data pool from which the report is to be generated, structured as a dictionary.
+        filter_content : Dict[str, Any]
+            A dictionary containing filter criteria, aggregation instructions, and scalar operation details.
+
+        Returns
+        -------
+        List[float]
+            The aggregated report data as a list.
+
+        Raises
+        ------
+        ValueError
+            This type of error is raised in the following cases:
+                If the report data is empty.
+                If the type of aggregation is not supported. The supported types are: sum, average, product,
+                    subtraction, division, SD.
+                If there are no horizontal or vertical aggregation keys in the filter content.
```

Comment:
```suggestion
        ValueError
            If the report data is empty.
            If the type of aggregation is not supported. The supported types are: sum, average, product,
                subtraction, division, SD.
            If there are no horizontal or vertical aggregation keys in the filter content.
```
This keeps the style consistent with what is requested in issue #1074